esphome/components/light/color_mode.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <cstdint>
+
+namespace esphome {
+namespace light {
+
+/// Color channels are the various outputs that a light has and that can be independently controlled by the user.
+enum class ColorChannel : uint8_t {
+  /// Intensity of a single white output.
+  WHITE = 1,
+  /// Color temperature of a single white output.
+  COLOR_TEMPERATURE = 2,
+  /// Intensities of a cold white and warm white output.
+  COLD_WARM_WHITE = 4,
+  /// Colored output using RGB format.
+  RGB = 8
+};
+
+// Define unary * operator to convert enum to underlying value -- this saves a lot of typing and ugly code.
+inline constexpr unsigned operator*(ColorChannel val) { return static_cast<uint8_t>(val); }
```

Comment:
```suggestion
inline constexpr uint8_t operator*(ColorChannel val) { return static_cast<uint8_t>(val); }
```

Would be more correct - let me know if there's a reason/only unsigned works.

Reply:
No, you're right, this was just me being inconsistent.